src/hooks/common/usePendingState.ts

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+const usePendingState = <T>() => {
+  const [pendingItem, setPendingItem] = React.useState<T>();
```

Comment:
wonder if this should be an array, so we can have pendingItems (which may be only one item in some instances, like editing a source... but could also work for when we are selecting several items, if that makes sense)

Reply:
I like the idea, but this change will break the code in a million parts. Let's add this in another pr.